src/shims/unix/fs.rs

```diff
@@ -1659,6 +1666,138 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn mkstemp(&mut self, template_op: &OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        use rand::distributions::{Distribution, Uniform};
+
+        let this = self.eval_context_mut();
+
+        let target_is_windows =
+            this.eval_context_ref().tcx.sess.target.families.contains(&Cow::Borrowed("windows"));
+
+        // Get the raw bytes from the template. This is passed to us from `std` as an `OsString`.
+        // These bytes represent a string in _target_ OS encoding, which may or may not match
+        // the _host_ OS encoding.
+        let template_bytes = this.eval_context_ref().read_c_str(this.read_pointer(template_op)?)?;
```

Comment:
Can this contain directory separators? If yes, you should use `read_path_from_c_str` instead.

Reply:
No it does not.